src/RpcClient.ts

```diff
@@ -262,54 +262,58 @@ export class RedirectRpcClient extends RpcClient {
         const message = UrlRpcEncoder.receiveRedirectResponse(window.location);
         if (message) {
             this._receive(message);
-
-        // The URL the user goes back to in the browser history (the page
-        // that this RpcClient is inited on) may itself have been a
-        // URL-encoded RPC request. Thus before triggering a potential
-        // rejection of the called request, we make sure that there is
-        // no RPC request in the URL, to be able to re-start the actual
-        // request that the user goes back to.
-        } else if (!UrlRpcEncoder.receiveRedirectCommand(window.location)) {
+        } else {
             this._rejectOnBack();
         }
     }
 
     /* tslint:disable-next-line:no-empty */
     public close() { }
 
-    public call(returnURL: string, command: string, ...args: any[]) {
-        this.callAndSaveLocalState(returnURL, null, command, ...args);
+    public call(returnURL: string, command: string, handleHistoryBack = false, ...args: any[]) {
+        this.callAndSaveLocalState(returnURL, null, command, handleHistoryBack, ...args);
     }
 
-    public callAndSaveLocalState(returnURL: string, state: ObjectType | null, command: string, ...args: any[]) {
+    public callAndSaveLocalState(returnURL: string, state: ObjectType | null, command: string, handleHistoryBack = false, ...args: any[]) {
         const id = RandomUtils.generateRandomId();
         const url = UrlRpcEncoder.prepareRedirectInvocation(this._target, id, returnURL, command, args);
 
         this._waitingRequests.add(id, command, state);
 
-        history.replaceState({rpcRequestId: id}, document.title);
+        if(handleHistoryBack) {
+            /**
+             * The rpcRequestId in the history.state is used to detect in the client
+             * if a history entry was visited before, which makes it a history.back
+             * navigation. The stored ID is then also used to retrieve the correct
+             * stored callback and waiting request, to be able to reject it.
+             */
+            history.replaceState({rpcRequestId: id}, '');
```

Comment:
Why is `''` better than `document.title`?

For better readability / understandability I would suggest to either: 
- store `rpcRequestId` and a `rejectOnBack` flag
- rename `rpcRequestId` to `rpcBackRejectionId`
- only store a `rejectOnBack` flag and read the id from the `_waitingRequests` stack (if possible)